dataaccess.js

```diff
@@ -364,18 +379,111 @@ function getPlayerRankings(playerId, showPersonalRank) {
     });
 }
 
-function calculateAllStats() {
-    debug('Calculating stats for every player');
+//This could be a bit dicey, so beware
+function updateResults(gameData) {
+    debug('Updating results for finished game');
+
+    for (let player of gameData.playerRank) {
+        if (player === 'ai') {
+            continue;
+        }
+
+        if (!stats[player]) {
+            //first time player
+            stats[player] = {
+                games: 0,
+                rank: 0,
+                playerName: '',
+                wins: 0,
+                winsAI: 0,
+                percent: 0
+            }
+        }
+
+        if (gameData.playerRank[0] === player) {
+            //This was the winner
+            stats[player].wins++;
+            if (gameData.humanPlayers < 2) {
+                //This was an ai game so it counts towards ai wins
+                stats[player].winsAI++;
+            }
+        }
+
+        stats[player].games++;
+        stats[player].percent = Math.floor(100 * stats[player].wins / stats[player].games);
+
+        //Make sure we have the player's name if it was a new player
+        if (!stats[player].playerName) {
```

Comment:
This is async, so the document will probably be saved below before the player name has actually been updated.

Will this case ever occur? Isn't a player name always assigned before that player can start or join a game?